recipes/yarp/meta.yaml

```diff
@@ -0,0 +1,130 @@
+{% set name = "yarp" %}
+{% set namecxx = "yarp-cxx" %}
+{% set namepython = "yarp-python" %}
+{% set version = "3.7.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/robotology/yarp/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3b2275e0cc17ce41e44f6ae15adfc76e49a61f5ead8a6dd42ce2cd88fffdcbed
+
+build:
+  number: 0
+
+outputs:
+  - name: {{ namecxx }}
+    script: build_cxx.sh  # [unix]
+    script: bld_cxx.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage(namecxx, max_pin='x.x.x') }}
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        # Visual Studio 2017 is not supported
+        - vs2019_win-64  # [win]
+        - cmake
+        - pkg-config
+        - ninja
+        - multisheller
+        - {{ cdt('mesa-libgl-devel') }}  # [linux]
+        - {{ cdt('mesa-dri-drivers') }}  # [linux]
+        - {{ cdt('libselinux') }}  # [linux]
+        - {{ cdt('libxdamage') }}  # [linux]
+        - {{ cdt('libxxf86vm') }}  # [linux]
+        - {{ cdt('libxext') }}  # [linux]
+        - {{ cdt('libxcb') }}  # [linux]
+        - {{ cdt('libxfixes') }}  # [linux]
+        - {{ cdt('libxau') }}  # [linux]
+        - {{ cdt('expat') }}  # [linux]
+
+      host:
+        - ace
+        - tinyxml
+        - eigen
+        - sdl
+        - sdl2
+        - sqlite
+        - jpeg
+        - bash-completion  # [not win]
+        - libi2c  # [linux]
+        - ycm-cmake-modules
+        - robot-testing-framework
+        - xorg-libxfixes  # [linux]
+        - libpng
+        - libzlib
+        - soxr
+        - libedit  # [not win]
+
+      run:
+        - ycm-cmake-modules
+        - eigen
```

Comment:
If you use YARP as a C++ library, Eigen is used in the public headers. This means that if you do not have installed `ycm-cmake-modules` and `eigen` your CMake project's call to `find_package(YARP REQUIRED)` will fail, even if you installed `mamba install yarp-cxx` in your environment. This is the reason why `ycm-cmake-modules` and `eigen` have been added as run dependencies, even if they are not strictly speaking run dependencies (for example if you use just the python bindings of YARP, you do not need them). So, it is basically a trade-off between convenience for direct C++ users of YARP as opposed to having "useless" packages installed for indirect C++ users of YARP and users of Python bindings of YARP. Given that eigen and ycm-cmake-modules are fairly small I favored the convenience for C++ users, but I am also happy to remove them if you prefer (we can always add a `yarp-cxx-devel` package in the future that depends on `yarp-cxx` and also on this dependencies).

Reply:
> > > The run_exports directive should be on the combined package yarp or only in the yarp-cxx package? For the time being I only put them in yarp-cxx to avoid the duplication of the max_pin information.
> > 
> > 
> > The run_export goes in yarp-cxx because that has the compiled-language API. Because yarp (combined) depends on yarp-cxx, if someone puts yarp in the host section, yarp-cxx will also be added and the run_export will be applied.
> 
> Oh, great, I did not know that the `run_exports` directive is transitive. Initially I tought this was not the case as otherwise downstream packages that depended on `yarp-python` gotan ABI-based constraint on yarp-cxx, even if they are not actually compiled packages, but then I realized that if you have a python package that depends on `yarp-python`, you will add it to `run` dependencies, not `host` one.

Actually, I'm wrong. [The docs specifically state:](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html)

> run_exports only affects directly named dependencies.

So if you want the run_export for when people add yarp instead of yarp-cxx to the host section of their package, you will need to add the run_export to the yarp meta-package.